File: lab2no1.py
print('Введите m') 
m = int(input())
print('Введите n')
n = int(input())
mCurr = nCurr = nStart = mStart = 1 
print('Введите последовательность команд')
line = input()
for i in line:
    if m < 1 or n < 1:
        print('Error')
        break
    else:
        if i == 'L':
            mCurr -= 1
        elif i == 'R':
            mCurr += 1
        elif i == 'D':
            nCurr += 1
        elif i == 'U':
            nCurr -= 1
if mCurr > m or nCurr > n:
    print('Error')
elif mCurr < 1 or nCurr < 1:
    if mCurr < 1:
        mStart = 2 + abs(mCurr)
    if nCurr < 1:
        nStart = 2 + abs(nCurr)
    print(mStart, nStart)
else:
    print('1 1')

File: lab2no1.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

let m, n, mStart, nStart;
let line = '';

rl.question('Enter m: \n', (input) => {
    m = parseInt(input);
    rl.question('Enter n: \n', (input) => {
        n = parseInt(input);
        rl.question('Enter line: \n', (input) => {
            line = input.trim();
            
            let mCurr = 1, nCurr = 1;
            
            if (m < 1 || n < 1) {
                console.log("Error");
            } else {
                for (let i = 0; i < line.length; i++) {
                    switch (line[i]) {
                        case 'L':
                            mCurr -= 1;
                            break;
                        case 'R':
                            mCurr += 1;
                            break;
                        case 'D':
                            nCurr += 1;
                            break;
                        case 'U':
                            nCurr -= 1;
                            break;
                        default:
                            break;
                    }
                }

                if (mCurr > m || nCurr > n) {
                    console.log("Error");
                } else if (mCurr < 1 || nCurr < 1) {
                    if (mCurr < 1) {
                        mStart = 2 + Math.abs(mCurr);
                    }
                    if (nCurr < 1) {
                        nStart = 2 + Math.abs(nCurr);
                    }
                    console.log(mStart + ' ' + nStart);
                } else {
                    console.log("1 1");
                }
            }
            
            rl.close();
        });
    });
});
